sys/linux/sys.txt

```diff
@@ -542,6 +549,38 @@ openat$bsg(fd const[AT_FDCWD], file ptr[in, string["/dev/bsg"]], flags flags[ope
+openat2$keychord(fd const[AT_FDCWD], file ptr[in, string["/dev/keychord"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$zygote(fd const[AT_FDCWD], file ptr[in, string["/dev/socket/zygote"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$sw_sync(fd const[AT_FDCWD], file ptr[in, string["/dev/sw_sync"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$pktcdvd(fd const[AT_FDCWD], file ptr[in, string["/dev/pktcdvd/control"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$lightnvm(fd const[AT_FDCWD], file ptr[in, string["/dev/lightnvm/control"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$sequencer(fd const[AT_FDCWD], file ptr[in, string["/dev/sequencer"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$sequencer2(fd const[AT_FDCWD], file ptr[in, string["/dev/sequencer2"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$xenevtchn(fd const[AT_FDCWD], file ptr[in, string["/dev/xen/evtchn"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$dlm_control(fd const[AT_FDCWD], file ptr[in, string["/dev/dlm-control"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$dlm_monitor(fd const[AT_FDCWD], file ptr[in, string["/dev/dlm-monitor"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$dlm_plock(fd const[AT_FDCWD], file ptr[in, string["/dev/dlm_plock"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$btrfs_control(fd const[AT_FDCWD], file ptr[in, string["/dev/btrfs-control"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$ubi_ctrl(fd const[AT_FDCWD], file ptr[in, string["/dev/ubi_ctrl"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$cachefiles(fd const[AT_FDCWD], file ptr[in, string["/dev/cachefiles"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$ndctl0(fd const[AT_FDCWD], file ptr[in, string["/dev/ndctl0"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$nmem0(fd const[AT_FDCWD], file ptr[in, string["/dev/nmem0"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$nvram(fd const[AT_FDCWD], file ptr[in, string["/dev/nvram"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$ocfs2_control(fd const[AT_FDCWD], file ptr[in, string["/dev/ocfs2_control"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$nvme_fabrics(fd const[AT_FDCWD], file ptr[in, string["/dev/nvme-fabrics"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+openat2$bsg(fd const[AT_FDCWD], file ptr[in, string["/dev/bsg"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
+
+openat2$snapshot(fd const[AT_FDCWD], file ptr[in, string["/dev/snapshot"]], how ptr[in, open_how_const_mode[0]], size bytesize[how]) fd
```

Comment:
For these syscalls I suspect certain resolution modes will fail since we have an absolute path or are using magic links.  Maybe we should limit the resolution mode?

Then again, there's only 5 modes, so maybe it doesn't matter much from efficiency point-of-view.

Reply:
Oh, we don't need openat2 for all devices. The goal of openat calls for devices is to test the devices, rather than the openat call itself (we test it with the generic version).
Using openat2 for devices will create portability problems for syzkaller reproducers. That's the reason why we use openat rather than simpler open: open is not present on arm64, so if we get a reproducer with open on x86_64, it won't work arm64. Openat2 is not present almost anywhere, so any reproducers with openat2 won't work almost anywhere.
And it just produces too much duplication (there are many more openat calls in other *.txt files as well).
Please remove the openat2 calls for devices.

Reply:
I see. Thanks for explaining this! I've just modified the commits correspondingly. I believe the procfs and pidfd part should be deleted as well for the same reason?